nixos/modules/programs/1password-gui.nix

```diff
@@ -0,0 +1,61 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+  cfg = config.programs._1password-gui;
+
+in
+{
+  options = {
+    programs._1password-gui = {
+      enable = mkOption {
+        default = false;
+        type = types.bool;
+        description = ''
+          Whether to install 1password-gui with systme-authentication.
+        '';
+      };
+    };
+  };
+
+  config = mkIf cfg.enable {
+    environment.systemPackages = [ pkgs._1password-gui ];
+
+    security = {
+      pam.services.polkit-1.fprintAuth = true;
+      polkit.enable = true;
+
+      wrappers = {
+        /*_1password-chrome-sandbox = {
+          source = "${pkgs._1password-gui.out}/share/1password/chrome-sandbox";
+          permissions = "u+rwx,g+rx,o-rx";
+        };*/
+
+        _1password-keyring-helper = {
+          # source = "${pkgs._1password-gui.out}/share/1password/1Password-KeyringHelper";
+          source = "${pkgs._1password-gui.out}/share/1password/1Password-KeyringHelper-Real";
+          owner = "root";
+          group = "onepassword";
+          # permissions = "u+rxs,g+rxs,o+rx";
+          # capabilities = "cap_setpcap+ep";
+          setuid = true;
+        };
+
+        _1password-browser-support = {
+          # source = "${pkgs._1password-gui.out}/share/1password/1Password-BrowserSupport";
+          source = "${pkgs._1password-gui.out}/share/1password/1Password-BrowserSupport-Real";
+          owner = "root";
+          group = "onepassword";
+          # permissions = "u+rxs,g+rxs,o+rx";
+          # capabilities = "cap_setpcap+ep";
+          setuid = true;
```

Comment:
Hi there,

This wrapper shouldn't be needed. The browser support binary has no special permissions:
![image](https://user-images.githubusercontent.com/68570223/128422417-75449d7e-40af-49d9-b826-f731fa48d81a.png)

Reply:
Thanks for the feedback, I removed the wrapper as I was unsure if it's needed. Do you have it working on NixOS or another distro? Do you know if browser-support does require working KeyringHelper because even without modifications (except needed patchelf) it does not unlock my browser-addon when desktop-tool is running.

Reply:
It does require one, yes. KeyringHelper is spawned as a subprocess of both browser-support and 1password. Therefore, it must be functional for either side to work. I have it working on Ubuntu and some other more "standard" distributions.

Reply:
Thanks for the feedback, than we will wait until 1password provides something new/fixed :) They have said at least, they try in the thread I joined/commented above :)

Reply:
What are the current errors you are getting in `~/.config/1password/logs/KeyringHelper/`? That seems like a good start.

Reply:
Yeah I understand, but if KeyringHelper is required it can't work as far as I understand, 1Password confirmed an issue (with an older release but haven't heard news) with Binary Verification when KeyringHelper can't be suid directly/is suid by wrapper like in NixOS.

Current relevant log entry from last Beta `8.1.2-82.BETA`:
`
WARN  2021-08-12T02:51:32.972 op_executor:invocation_loop(ThreadId(22)) [1P:foundation/op-linux/src/kernel_keyring.rs:824] failed to initialize keyring helper, its functionality will be unavailable: Io(Error { kind: UnexpectedEof, message: "failed to fill whole buffer" })
`

Old error for reference (from above):
`
ERROR 2021-07-19T20:39:11.506 main(ThreadId(1)) [1P:foundation/op-linux/src/bin/keyring_helper.rs:174] failed to verify keyring helper process permissions, aborting: BinaryPermissions
WARN 2021-07-19T20:39:11.506 op_executor:invocation_loop(ThreadId(20)) [1P:foundation/op-linux/src/kernel_keyring.rs:824] failed to initialize keyring helper, its functionality will be unavailable: Io(Error { kind: UnexpectedEof, message: "failed to fill whole buffer" })
`

Reply:
These look like logs from 1password itself, but not anything from the keyringhelper 🤔. What is inside `~/.config/1Password/logs/BrowserSupport/KeyringHelper`? Do you get any more info setting `OP_LOG_LEVEL=debug`?

Reply:
I'm getting a log output for KeyringHelper in `.config/1Password/logs/BrowserSupport/KeyringHelper/1Password_rCURRENT.log`:
`
INFO  2021-08-12T04:34:51.380 main(ThreadId(1)) [1P:foundation/op-linux/src/bin/keyring_helper.rs:144] initalizing keyring helper
ERROR 2021-08-12T04:34:51.384 main(ThreadId(1)) [1P:foundation/op-linux/src/bin/keyring_helper.rs:150]
                keyring helper detected it was not running as root. This could lead to credentials being compromised, aborting!
                Permissions found: EUID: 1000, EGID: 100
`

Maybe 1Password is running as User?

BrowserHelper-Log show that it's failing because of KeyringHelper:
`
WARN  2021-08-12T04:34:51.385 main(ThreadId(1)) [1P:foundation/op-linux/src/kernel_keyring.rs:824] failed to initialize keyring helper, its functionality will be unavailable: Io(Error { kind: UnexpectedEof, message: "failed to fill whole buffer" })
ERROR 2021-08-12T04:34:51.390 main(ThreadId(1)) [1P:native-messaging/op-browser-support/src/main.rs:72] CodecError(Linux(KeyNotFound))
	Happened in: native-messaging/op-browser-support/src/main.rs:235
INFO  2021-08-12T04:34:51.390 main(ThreadId(1)) [1P:native-messaging/op-browser-support/src/main.rs:88] Shutting down 1Password-BrowserSupport 8.1.2-82.BETA production build no. 80102082.
`

Reply:
It looks that way. This check is triggered when `geteuid() != 0`

Reply:
But running 1Password as root in general seems like no solution?! :D
Do you have a suggestion or is waiting for a patched release the only option?